subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/InstantExecutionIncludedBuildChangesIntegrationTest.groovy

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.instantexecution
+
+
+import org.gradle.api.provider.ValueSource
+import org.gradle.api.provider.ValueSourceParameters
+import org.gradle.instantexecution.fixtures.BuildLogicChangeFixture
+import spock.lang.Unroll
+
+import static org.junit.Assume.assumeFalse
+
+class InstantExecutionIncludedBuildChangesIntegrationTest extends AbstractInstantExecutionIntegrationTest {
+
+    @Unroll
+    def "invalidates cache upon change to included #fixtureSpec"() {
+        given:
+        def instant = newInstantExecutionFixture()
+        def fixture = fixtureSpec.fixtureForProjectDir(file('build-logic'))
+        fixture.setup()
+        settingsFile << """
+            includeBuild 'build-logic'
+        """
+        buildFile << """
+            plugins { id('$fixture.pluginId') }
+        """
+
+        when:
+        instantRunLenient fixture.task
+
+        then:
+        outputContains fixture.expectedOutputBeforeChange
+        instant.assertStateStored()
+
+        when:
+        fixture.applyChange()
+        instantRunLenient fixture.task
+
+        then:
+        outputContains fixture.expectedOutputAfterChange
+        instant.assertStateStored()
```

Comment:
It would be good to assert that the user is informed of why the cache entry is not reused

Same for all of these invalidation tests.

Reply:
Done in https://github.com/gradle/gradle/pull/14348.